crates/miden-tx/src/tests/kernel_tests/test_faucet.rs

```diff
@@ -620,6 +614,47 @@ fn test_burn_non_fungible_asset_fails_inconsistent_faucet_id() {
     assert_execution_error!(process, ERR_FAUCET_NON_FUNGIBLE_ASSET_TO_BURN_NOT_FOUND);
 }
 
+// IS NON FUNGIBLE ASSET ISSUED TESTS
+// ================================================================================================
+
+#[test]
+fn test_is_non_fungible_asset_issued_succeeds() {
+    let tx_context = TransactionContextBuilder::with_non_fungible_faucet(
+        NonFungibleAsset::mock_issuer().into(),
+        ONE,
+        false,
+    )
+    .build();
+
+    let non_fungible_asset_1 = NonFungibleAsset::mock(&NON_FUNGIBLE_ASSET_DATA);
+    let non_fungible_asset_2 = NonFungibleAsset::mock(&NON_FUNGIBLE_ASSET_DATA_2);
+
+    let code = format!(
+        "
+        use.kernel::prologue
+        use.miden::faucet
+
+        begin
+            exec.prologue::prepare_transaction
+
+            # check that NON_FUNGIBLE_ASSET_DATA_2 is already issued
+            push.{non_fungible_asset_2}
+            exec.faucet::is_non_fungible_asset_issued
+            eq.1 assert
+
+            # check that NON_FUNGIBLE_ASSET_DATA was not issued yet
+            push.{non_fungible_asset_1}
+            exec.faucet::is_non_fungible_asset_issued
+            eq.0 assert
```

Comment:
I think when writing MASM tests with assertions, I would prefer including at least some unique error code for every assertion, e.g. `assert.err=9999` so that when it fails, it's a bit easier to debug where it is.